omegaconf/_utils.py

```diff
@@ -234,10 +235,17 @@ def extract_dict_subclass_data(obj: Any, parent: Any) -> Optional[Dict[str, Any]
     """Check if obj is an instance of a subclass of Dict. If so, extract the Dict keys/values."""
     from omegaconf.omegaconf import _maybe_wrap
 
+    obj_type = type(obj)
+    if is_dict_subclass(obj) or is_dict_subclass(obj_type):
+        warnings.warn(
+            "Subclassing `Dict` in Structured Config classes is deprecated, see github.com/omry/omegaconf/issues/663",
+            UserWarning,
+            stacklevel=1,
+        )
+
```

Comment:
will moving the warnings to block starting at 249 work?

Reply:
It doesn't work -- the warnings need to come before the `return None` on line 247.